langkit/templates/pkg_implementation_body_ada.mako

```diff
@@ -354,12 +356,13 @@ package body ${ada_lib_name}.Implementation is
    is
       Actual_Charset : constant String :=
         (if Charset = "" then Default_Charset else Charset);
-      Symbols        : constant Symbol_Table := Create_Symbol_Table;
+      Symbols        : constant Precomputed_Symbol_Table
+        := Create_Symbol_Table;
       Context        : Internal_Context;
    begin
       Context_Pool.Acquire (Context);
       Context.Ref_Count := 1;
-      Context.Symbols := Symbols;
+      Context.Symbols := Symbol_Table (Symbols);
```

Comment:
It looks like some users of Context.Symbols need the Precomputed_Symbol_Table while others could do with it as well, so what about changing its type, to avoid the multiple conversions here and there?

Reply:
Because most of the functions on the `Symbol_Table` are non dispatching, and are defined in such a way that a cast would be needed to call them if we stored a `Precomputed_Symbol_Table` instead. Hence we favor the common use case.

Reply:
Ok I see. Go ahead, thanks!